lib/chat/video_view.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+class VideoView extends StatefulWidget {
+  VideoView({
+    this.videoToPlay = "",
+    this.videoFromWebToPlay = ""});
+
+  final String videoToPlay;
+  final String videoFromWebToPlay;
+
+  @override
+  _VideoViewState createState() => _VideoViewState();
+}
+
+class _VideoViewState extends State<VideoView> {
+  VideoPlayerController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget.videoFromWebToPlay != null && widget.videoFromWebToPlay.isNotEmpty) {
+      _controller = VideoPlayerController.network(widget.videoFromWebToPlay);
+    } else if (widget.videoToPlay != null && widget.videoToPlay.isNotEmpty) {
+      _controller = VideoPlayerController.file(File(widget.videoToPlay));
+    }
+    _controller?.addListener(() {
+      setState(() {});
+    });
+    if (_controller?.value?.initialized == false) {
+      _controller.initialize().then((_) {
+        setState(() {});
+      });
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _controller.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_controller == null || _controller?.value?.hasError == true) {
+      return Center(
+        child: Text(_controller?.value?.errorDescription ?? ""),
+      );
+    }
+    if (!_controller.value.initialized) {
+      return Center(child: CircularProgressIndicator());
+    } else {
+      return Container(
+        color: Colors.black,
+        child: Column(
+          mainAxisSize: MainAxisSize.max,
+          children: [
+            Expanded(
+              flex: 8,
+              child: AspectRatio(
+                aspectRatio: _controller.value.aspectRatio,
+                child: Stack(alignment: Alignment.bottomCenter, children: [
+                  VideoPlayer(_controller),
+                  _PlayPauseOverlay(controller: _controller),
+                  VideoProgressIndicator(_controller, allowScrubbing: true),
+                ]),
+              ),
+            ),
+          ],
+        ),
+      );
+    }
+  }
+}
+
+class _PlayPauseOverlay extends StatefulWidget {
+  const _PlayPauseOverlay({Key key, this.controller}) : super(key: key);
+
+  final VideoPlayerController controller;
+
+  @override
+  __PlayPauseOverlayState createState() => __PlayPauseOverlayState();
+}
+
+class __PlayPauseOverlayState extends State<_PlayPauseOverlay> {
```

Comment:
Why do you need two _ ?

Reply:
Just autogenerated, because widget have single underline and it appended to state. 
Removed redundant underline.